src/js/polyfills/webvtt.js

```diff
@@ -696,16 +715,26 @@ function moveBoxToLinePosition(window, styleBox, containerBox, boxPositions, num
             const p = b.intersectPercentage(containerBox);
             // If we're outside the container box less then we were on our last try
             // then remember this position as the best position.
-            if (percentage > p) {
+            if (percentage <= p) {
                 bestPosition = new BoxPosition(b);
                 percentage = p;
+                bestAxis = axis[i];
             }
             // Reset the box position to the specified position.
             b = new BoxPosition(specifiedPosition);
         }
-        return bestPosition || specifiedPosition;
-    }
 
+        let finalPos = bestPosition || specifiedPosition;
+        // If we detect a best position based on one axis, oftentimes combining with the other axis
+        // results in a much better position.
+        // TODO: Make more performant by checking axis permutations above
```

Comment:
there is certainly a way to refactor this to be O(n) however the complexity mixed with the amount of time already sunk into this make me believe this option is the best way forward until we want to spend more time on this rendering